core/src/scene/sceneLoader.cpp

```diff
@@ -528,6 +525,51 @@ Node SceneLoader::mixStyle(const Mixes& mixes) {
     return styleNode;
 }
 
+Mixes SceneLoader::recursiveMixins(Mixes mixIn, const std::string& styleName, Node styles, UniqueStyles& uniqueStyles) {
+    Mixes mixes;
+    mixes.swap(mixIn);
+
+    static std::unordered_set<std::string> mixedStyles;
```

Comment:
This is gonna be a problem if scene loading is ever performed more than once; entries will remain in this set for the entire duration of the program execution.